ironfish-cli/src/commands/wallet/prune.ts

```diff
@@ -27,10 +31,52 @@ export default class PruneCommand extends IronfishCommand {
     CliUx.ux.action.start(`Opening node`)
     const node = await this.sdk.node()
     await NodeUtils.waitForOpen(node)
-    await node.wallet.open()
-    await node.wallet.walletDb.open()
     CliUx.ux.action.stop('Done.')
 
+    if (!node.chain.synced) {
+      this.log(
+        `Your node must be synced with the Iron Fish network to prune wallet transactions. Please try again later`,
+      )
+      this.exit(1)
+    }
+
+    const walletHeadHash = await node.wallet.getLatestHeadHash()
+
+    if (walletHeadHash === null) {
+      this.log(`Failed to get latest head hash.`)
+      this.exit(1)
+      return
+    }
+
+    const head = await node.chain.getHeader(walletHeadHash)
+
+    if (head === null) {
+      this.log(`Failed to get chain header.`)
+      this.exit(1)
+      return
+    }
+
+    for (const account of node.wallet.listAccounts()) {
+      let count = 0
+
+      for await (const { transaction } of account.getExpiredTransactions(head.sequence)) {
+        count = +1
+        this.log(
+          `Account ${account.displayName} has expired transaction with hash ${transaction
+            .hash()
+            .toString('hex')}`,
+        )
```

Comment:
this will be printed out for every single transactions which could be thousands of them. Dso we need this printed out every time?